pkg/server/backend_manager.go

```diff
@@ -121,17 +120,39 @@ func (b *backend) Context() context.Context {
 	return b.conn.Context()
 }
 
-func newBackend(conn agent.AgentService_ConnectServer) *backend {
+func (b *backend) GetAgentIdentifiers() (header.Identifiers, error) {
+	var agentIdentifiers header.Identifiers
+	md, ok := metadata.FromIncomingContext(b.Context())
+	if !ok {
+		return agentIdentifiers, fmt.Errorf("failed to get context")
+	}
+	agentIDs := md.Get(header.AgentIdentifiers)
+	if len(agentIDs) > 1 {
+		return agentIdentifiers, fmt.Errorf("expected at most one agent IP in the context, got %v", agentIDs)
+	}
+	if len(agentIDs) == 0 {
```

Comment:
nit
```suggestion
	else if len(agentIDs) == 0 {
```

Reply:
Note this was only a code move, let's not change the logic in the same PR as this bugfix.